File: A_Password_Generator.kt
fun generatePassword(a: Int, b: Int, c: Int): String {
    val digits = "0123456789"
    val upper = "ZYXWVUTSRQPONMLKJIHGFEDCBA"
    val lower = "abcdefghijklmnopqrstuvwxyz"

    val sb = StringBuilder()
    var di = 0
    var ui = 0
    var li = 0

    var aCount = a
    var bCount = b
    var cCount = c

    while (aCount + bCount + cCount > 0) {
        if (aCount > 0 && (sb.isEmpty() || sb.last() != digits[di])) {
            sb.append(digits[di])
            di = (di + 1) % 10
            aCount--
        }
        if (bCount > 0 && (sb.isEmpty() || sb.last() != upper[ui])) {
            sb.append(upper[ui])
            ui = (ui + 1) % 26
            bCount--
        }
        if (cCount > 0 && (sb.isEmpty() || sb.last() != lower[li])) {
            sb.append(lower[li])
            li = (li + 1) % 26
            cCount--
        }
    }

    return sb.toString()
}

fun main() {
    val t = readln().toInt()
    repeat(t) {
        val (a, b, c) = readln().split(" ").map { it.toInt() }
        println(generatePassword(a, b, c))
    }
}

File: A_Password_Generator.cpp
#include <bits/stdc++.h>
using ll = long long;

std::string generate_password(int a, int b, int c) {
    std::string digits = "0123456789";
    std::string upper = "ZYXWVUTSRQPONMLKJIHGFEDCBA";
    std::string lower = "abcdefghijklmnopqrstuvwxyz";

    std::string str;
    int di = 0, ui = 0, li = 0;

    while (a + b + c > 0) {
        if (a > 0 && (str.empty() || str.back() != digits[di])) {
            str += digits[di];
            di = (di + 1) % 10;
            a--;
        }
        if (b > 0 && (str.empty() || str.back() != upper[ui])) {
            str += upper[ui];
            ui = (ui + 1) % 26;
            b--;
        }
        if (c > 0 && (str.empty() || str.back() != lower[li])) {
            str += lower[li];
            li = (li + 1) % 26;
            c--;
        }
    }

    return str;
}

void solve() {
    int a, b, c;
    std::cin >> a >> b >> c;
    std::cout << generate_password(a, b, c) << "\n";
}

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int t;
    std::cin >> t;
    while (t--) {
        solve();
    }
}
